tests/DotNetLightning.Core.Tests/TransactionTests.fs

```diff
@@ -326,4 +329,551 @@ let testList = testList "transaction tests" [
+            StaticChannelConfig = staticRemoteChannelConfig
+            RemotePerCommitmentSecrets = remoteRemotePerCommitmentSecrets
+            ShortChannelId = None
+            LocalCommit = remoteLocalCommit
+            RemoteCommit = remoteRemoteCommit 
+            LocalChanges = LocalChanges.Zero
+            RemoteChanges = RemoteChanges.Zero
+        }
+
+        let remoteSpendingLocalCommitmentRes =
+            ClosingHelpers.HandleFundingTxSpent
+                remoteSavedChannelState
+                None
+                remoteChannelPrivKeys
+                localCommitmentTx
+
+        let outputsSumFromRemoteSpendingLocalCommitment =
+            (Result.deref remoteSpendingLocalCommitmentRes.MainOutput)
+                .SendAll(remoteDestPubKey.WitHash.ScriptPubKey)
+                .SendFees(Money.Zero)
+                .BuildTransaction(true)
+                .Outputs
+                |> Seq.sumBy (fun output -> output.Value)
+            
+        Expect.equal outputsSumFromRemoteSpendingLocalCommitment (remoteAmount.ToMoney()) "wrong to_remote spending tx"
+
+        let remoteSpendingRemoteCommitmentRes =
+            ClosingHelpers.HandleFundingTxSpent
+                remoteSavedChannelState
+                None
+                remoteChannelPrivKeys
+                remoteCommitmentTx
+
+        let outputsSumFromRemoteSpendingRemoteCommitment =
+            (Result.deref remoteSpendingRemoteCommitmentRes.MainOutput)
+                .SendAll(remoteDestPubKey.WitHash.ScriptPubKey)
+                .SendFees(Money.Zero)
+                .BuildTransaction(true)
+                .Outputs
+                |> Seq.sumBy (fun output -> output.Value)
+            
+        Expect.equal outputsSumFromRemoteSpendingRemoteCommitment (remoteAmount.ToMoney()) "wrong to_local spending tx"
+
+        let remoteSpendingRevokedRemoteCommitmentRes =
+            ClosingHelpers.HandleFundingTxSpent
+                remoteSavedChannelState
+                None
+                remoteChannelPrivKeys
+                revokedLocalCommitmentTx
+
+        let outputsSumFromRemoteSpendingRevokedRemoteCommitment =
+            (Result.deref remoteSpendingRevokedRemoteCommitmentRes.MainOutput)
+                .SendAll(remoteDestPubKey.WitHash.ScriptPubKey)
+                .SendFees(Money.Zero)
+                .BuildTransaction(true)
+                .Outputs
+                |> Seq.sumBy (fun output -> output.Value)
+            
+        let expectedAmountFromToLocal =
+            let localAmount = lastLocalCommitmentSpec.ToLocal.ToMoney()
+            let fee = revokedLocalCommitmentTx.GetFee [| fundingScriptCoin :> ICoin |]
+            localAmount - fee
+        let expectedAmountFromToRemote =
+            lastLocalCommitmentSpec.ToRemote.ToMoney()
+
+
+        Expect.equal outputsSumFromRemoteSpendingRevokedRemoteCommitment (expectedAmountFromToRemote + expectedAmountFromToLocal) "wrong penalty tx"
```

Comment:
You are running 3 attempt for `HandleFundingTxSpent`, and setup and assertions are mostly the same, it can be more succinct.

Reply:
I grouped validation, commitment tx creation (as much as I could) into functions